### PR TITLE
Port latest docs and inheritdocs

### PR DESCRIPTION
**Pull request by carlossanlop**
This should contain the majority of the undoc APIs introduced in 7.0.

### CODE REVIEW

**Review by rzikm — APPROVED** [review]
System.Net.* LGTM

File: xml/System.Security.Cryptography.X509Certificates/CertificateRequest.xml [hunk 1]
```diff
@@ -828,6 +828,30 @@ When submitting a certificate signing request via a web browser or other graphic
         <exception cref="T:System.ArgumentNullException">
           <paramref name="signatureGenerator" /> is <see langword="null" />.</exception>
         <exception cref="T:System.Security.Cryptography.CryptographicException">A cryptographic error occurs while creating the signing request.</exception>
+        <exception cref="T:System.InvalidOperationException">
+          <see cref="P:System.Security.Cryptography.X509Certificates.CertificateRequest.OtherRequestAttributes" /> contains a <see langword="null" /> value.
+
+-or-
+
+<see cref="P:System.Security.Cryptography.X509Certificates.CertificateRequest.OtherRequestAttributes" /> contains an entry with a <see langword="null" /><see cref="P:System.Security.Cryptography.AsnEncodedData.Oid" /> value.
```
Comment (gewarren) [inline xml/System.Security.Cryptography.X509Certificates/CertificateRequest.xml:836]:
```suggestion
<see cref="P:System.Security.Cryptography.X509Certificates.CertificateRequest.OtherRequestAttributes" /> contains an entry with a <see langword="null" /> <see cref="P:System.Security.Cryptography.AsnEncodedData.Oid" /> value.
```

File: xml/System.Security.Cryptography.X509Certificates/CertificateRequest.xml [hunk 2]
```diff
@@ -828,6 +828,30 @@ When submitting a certificate signing request via a web browser or other graphic
...
+<see cref="P:System.Security.Cryptography.X509Certificates.CertificateRequest.OtherRequestAttributes" /> contains an entry representing the PKCS#9
+                Extension Request Attribute (1.2.840.113549.1.9.14).
+
+-or-
+
+<see cref="P:System.Security.Cryptography.X509Certificates.CertificateRequest.CertificateExtensions" /> contains a <see langword="null" /> value.
+
+-or-
+
+<see cref="P:System.Security.Cryptography.X509Certificates.CertificateRequest.CertificateExtensions" /> contains an entry with a <see langword="null" /><see cref="P:System.Security.Cryptography.AsnEncodedData.Oid" /> value.
```
Comment (gewarren) [inline xml/System.Security.Cryptography.X509Certificates/CertificateRequest.xml:849]:
```suggestion
<see cref="P:System.Security.Cryptography.X509Certificates.CertificateRequest.CertificateExtensions" /> contains an entry with a <see langword="null" /> <see cref="P:System.Security.Cryptography.AsnEncodedData.Oid" /> value.
```

File: xml/System.Security.Cryptography.X509Certificates/CertificateRequest.xml [hunk 3]
```diff
@@ -828,6 +828,30 @@ When submitting a certificate signing request via a web browser or other graphic
...
+
+-or-
+
+This object was created with a constructor which did not accept a signing key.
```
Comment (gewarren) [inline xml/System.Security.Cryptography.X509Certificates/CertificateRequest.xml:853]:
```suggestion
This object was created with a constructor that did not accept a signing key.
```

**Review by gewarren — APPROVED** [review]

**Comment by carlossanlop** [comment]
IIRC, I have to remove OtherRequestAttributes again. It's not available in .NET 7, but is being ported because the porting tool took the comments from `main` in runtime.

File: xml/System.Runtime.Serialization/ImportOptions.xml [hunk 4]
```diff
@@ -107,7 +107,7 @@
       <Docs>
         <summary>Gets or sets a data contract surrogate that can be used to modify the code generated during an import operation.</summary>
         <value>An implementation of the <see cref="T:System.Runtime.Serialization.IDataContractSurrogate" /> interface that handles schema import.</value>
-        <remarks>To be added.</remarks>
+        <remarks>The interface type for this option is ISerializationSurrogateProvider, but to take full advantage of the imported code modification abilities, using an <see cref="T:System.Runtime.Serialization.ISerializationSurrogateProvider2" />ISerializationSurrogateProvider2 that also implements <see cref="T:System.Runtime.Serialization.ISerializationCodeDomSurrogateProvider" /> is recommended.</remarks>
```
Comment (carlossanlop) [inline xml/System.Runtime.Serialization/ImportOptions.xml:110]: The `ISerializationSurrogateProvider2` API will be available in RC2. The warning will go away after we update the mdoc ref assemblies to RC2.

https://github.com/dotnet/runtime/blob/release/7.0/src/libraries/System.Runtime.Serialization.Primitives/src/System/Runtime/Serialization/ISerializationSurrogateProvider2.cs

File: xml/System.Security.Cryptography.X509Certificates/CertificateRequest.xml [hunk 5]
```diff
@@ -828,6 +828,30 @@ When submitting a certificate signing request via a web browser or other graphic
         <exception cref="T:System.ArgumentNullException">
           <paramref name="signatureGenerator" /> is <see langword="null" />.</exception>
         <exception cref="T:System.Security.Cryptography.CryptographicException">A cryptographic error occurs while creating the signing request.</exception>
+        <exception cref="T:System.InvalidOperationException">
+          <see cref="P:System.Security.Cryptography.X509Certificates.CertificateRequest.OtherRequestAttributes" /> contains a <see langword="null" /> value.
```
Comment (carlossanlop) [inline xml/System.Security.Cryptography.X509Certificates/CertificateRequest.xml:832]: Actually, the `OtherRequestAttributes` API will also be available in RC2, and so the CI warning will go away when we bump the ref assemblies.

**Review by carlossanlop — COMMENTED** [review]
@gewarren we can ignore the two warnings about unavailable APIs.

**Comment by carlossanlop** [comment]
Merging with the warnings. They will go away when we update mdoc to RC1.

**Pull request merged**